Microsoft.Toolkit.Uwp.UI/AdvancedCollectionView/AdvancedCollectionView.cs

```diff
@@ -677,33 +677,17 @@ private bool HandleItemAdded(int newStartingIndex, object newItem, int? viewInde
                     return false;
                 }
 
-                if (newStartingIndex == 0 || _view.Count == 0)
-                {
-                    newViewIndex = 0;
-                }
-                else if (newStartingIndex == _source.Count - 1)
-                {
-                    newViewIndex = _view.Count - 1;
-                }
-                else if (viewIndex.HasValue)
+                if (viewIndex.HasValue)
                 {
                     newViewIndex = viewIndex.Value;
                 }
                 else
                 {
-                    for (int i = 0, j = 0; i < _source.Count; i++)
-                    {
-                        if (i == newStartingIndex)
-                        {
-                            newViewIndex = j;
-                            break;
-                        }
-
-                        if (_view[j] == _source[i])
-                        {
-                            j++;
-                        }
-                    }
+                    newViewIndex = _view.Select(x => _source.IndexOf(x)) // Get indexes of all items that are currently in view
```

Comment:
Definitely worried about the performance impact here, this seems like a lot of extra work and memory to determine this calculation.

Reply:
Yes it could impact on performance, but not sure about any other possible solutions right now. Someone should have to run benchmark stuff on this method to get a better overview.

Reply:
If the unit test you added shows the failure case at least, then it'd be good if we can figure out how to fix the original algorithm here. We wouldn't want to add extra overhead to this helper.

This class is relatively isolated, so you may be able to pull it out to run benchmarking on it in isolation vs. trying to add benchmarking within the project. @Sergio0694 any tips/resources/articles to point @w-ahmad to for benchmarking code like this?